### DISCUSSION TITLE
How to reset the com port?

### DISCUSSION THREAD

**Discussion by infromthecold** [maziac/DeZog#85] [Q&A] [answered]
Its great that I don't have to manually start the serial connection now, however sometimes I have to restart my next because it wont communicate anymore, however this is leaving the com port still connected so I am getting "Com port in use" type messages and terminating dezog leaves it still in use.

The only option is to restart visual studio code, which is annoying to say the least. 

How can i work around this so i don't have to restart  visual studio code?

**Comment by maziac** [comment]
Is this on Windows?
I have tested on Windows but only a few times. On the Mac I haven't seen this behavior.
If this is a windows thing I need to investigate.

How often is "sometimes"? And do you maybe have a way to reproduce.

**Comment by infromthecold** [comment]
Yes im on windows,

"No response received from remote. A simple reason for this message is that
the ZX Next is running the debugged program and cannot answer. In that case
press the yellow NMI button on the ZX Next to pause execution. It can, of
course, be also any other connection problem."

"Opening COM8: Access denied"

press the Nim with symbol shift on the next

"DeZog is already active."

Click terminate current session

press debug again

"DeZog is already active."

at this point i have to exit vs code

I wonder if we could have a command in VS code to terminate the com
connection from the debug console?

**Comment by infromthecold** [comment]
Also when trying to cancel using the terminate button i get

"Could not terminate active debug session. Please try manually."

**Comment by maziac** [comment, marked as answer]
Please try the fix here:
https://github.com/maziac/DeZog/releases/tag/v2.6.2-rc1

**Comment by infromthecold** [comment]
Great stuff, working a treat so far, thanks for the fix, not messing with a
separate serial connection makes life much easier, and it's even better now
that I can just click terminate to restart the connection.

Thanks again!
